### ISSUE TITLE
Override completion should place cursor near arrow

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#29495]
Currently, the cursor is placed at the end of the inserted code (as shown below).

At least in the case of a completion with an arrow body, that isn't most useful, since you won't type more statements in the body and you likely won't leave the inserted code as-is.

I'd propose that the cursor be left between the fat arrow and the inserted code.

This way, you could easily:
- convert the body to a block body,
- delete the body to type what you want.

![override-completion](https://user-images.githubusercontent.com/12466233/44597518-c5272a00-a784-11e8-8f87-15f545e4b586.gif)

Also, I think the override completion should not add a body when there is already one. This happens when you copy&paste one override method, then rename the method name.

**Comment by sharwell** [comment]
> I'd propose that the cursor be left between the fat arrow and the inserted code.

I like this because it enables both wrapping and Use Block Body scenarios. Though we normally wrap these before the arrow not after.

**Comment by CyrusNajmabadi** [comment]
> Though we normally wrap these before the arrow not after.

Note: we really should just have a formatting option for this.  :)       We were hesitant originally because it was a new language feature and we didn't know how it would be used.  At this point the major patterns are known and can be accounted for.

**Comment by Neme12** [comment]
Like @CyrusNajmabadi suggested, I would prefer if this just selected the expression in the body, because it's very likely that user will want to replace that.

> This way, you could easily:
> 
> * convert the body to a block body,
> * delete the body to type what you want.

If we put the caret before the arrow, I'm not sure deleting the body is so easy (it requires moving a few characters right, and then selecting the whole expression). Converting the body to a block body should always be easy using the refactoring: 
![image](https://user-images.githubusercontent.com/11444821/48034018-eaa9ba00-e15d-11e8-9661-45952346a49e.png)

**Comment by Neme12** [comment]
Actually, this issue mentions putting the caret to the right of the arrow:
> I'd propose that the cursor be left between the fat arrow and the inserted code.

I was looking at the PR when writing above comment, which implements care to the left of the arrow. I'm a little confused. @jcouv ?

**Comment by jcouv** [comment]
You're correct, the PR puts the caret to the left of the arrow, so that the second scenario you listed becomes easier (deleting the body). I forgot to update the issue to reflect this. Sorry for the confusion.

There are three scenarios considered so far:
1) insert breakline before arrow,
2) convert the body to a block body,
3) delete the body to type what you want.

And some options:
1) caret on left of arrow (PR)
2) caret on right of arrow (initially proposed in issue)
3) select the expression (alternative proposal)
4) caret on right of expression-statement (status quo)

I think option 3 improves the third scenario (delete the expression), but worsens the first scenario (insert breakline) and second scenario (refactoring to convert to block doesn't trigger).

**Comment by Neme12** [comment]
> (refactoring to convert to block doesn't trigger).

It does trigger, but you're right that the experience is worse because there are other additional refactorings and it's at the bottom of the list.

**Comment by jcouv** [comment]
> It does trigger

What selection did you use?

Here's what I get when selecting the expression:
![image](https://user-images.githubusercontent.com/12466233/48074771-89eaa380-e196-11e8-9f9c-0d76b5def38f.png)

Selecting the expression-statement:
![image](https://user-images.githubusercontent.com/12466233/48074806-9b33b000-e196-11e8-97be-4da647182215.png)

Selecting the arrow-clause:
![image](https://user-images.githubusercontent.com/12466233/48074839-ac7cbc80-e196-11e8-9b78-a90fe51e084a.png)

**Comment by Neme12** [comment]
I see. I think it's probably because in my case, it is the use block body analyzer (which is always offered anywhere on the line) as opposed to a refactoring, whereas you might have different settings in tools -> options:
![image](https://user-images.githubusercontent.com/11444821/48075255-19b63f00-e1e3-11e8-88ad-2e63f918dad2.png)

**Comment by jcouv** [comment]
From design meeting today:
- override completion should result in selecting the expression right-to-left (ie. caret on the left),
- convert to block body should trigger in that case.

**Comment by ToddGrun** [comment]
Should be fixed by https://github.com/dotnet/roslyn/pull/76983 and https://github.com/dotnet/roslyn/pull/76983
